scripts/getChangelog.py

```diff
@@ -1,90 +1,587 @@
+      print(f"\r{char} {state['text']}...                    ", end="", flush=True)
+      time.sleep(0.5)
+  print("\r                                       ", end="\r", flush=True)
+
+# Function: Get list of PRs for a given commit
+# Note: this might be unnecessary in case commits can only have one PR
+def get_associated_prs(commitid):
+  """
+  Called in case the number of PRs linked to a commit exceeds the maximum number of PRs that the script will fetch per commit.
+  """
+  global animation_text
+  animation_text = "Get PRs linked to commit"
+  query = '''
+  query ($cursor: String, $owner: String!, $repo: String!, $commitid: GitObjectID!) { 
+    repository(name: $repo, owner: $owner) { 
+      object(oid: $commitid) {
+        ... on Commit {
+          associatedPullRequests(first: 100, after: $cursor) {
+            pageInfo {
+              endCursor
+              hasNextPage
+            }
+            nodes {
+              number
+            }
+          }
+        }
+      }
+    }
+  }
+  '''
+  # Variable definition for GraphQL Explorer:
+  # {"commitid": "863d4c3b611df83389d82958114bfd2d1204e457", "cursor": null, "owner": "AntennaPod", "repo": "AntennaPod"}
+  has_next_pr_page = True
+  cursor = None
+  pr_numbers = []  # Create PR list
+  while has_next_pr_page:
+    variables = {"cursor": cursor, "owner": owner, "repo": repo, "commitid": commitid }
+    result = client.execute(query, variables)
+    data = json.loads(result)        
+    pr_numbers.extend([pr['number'] for pr in data['data']['repository']['commit']['associatedPullRequests']['nodes']])
+    page_info = data['data']['repository']['commit']['associatedPullRequests']['pageInfo']
+    has_next_pr_page = page_info['hasNextPage']
+    cursor = page_info['endCursor']
+    
+    time.sleep(1)  # To prevent hitting rate limits
+    return pr_numbers
+
+try: # Catch KeyboardInterrupt
+  # Define animation thread to avoid errors
+  animation_thread = None
+
+  # Define token
+  print(f"Hello, welcome to the {owner} PR list generator!")
+  time.sleep(0.5)
+  print("First, please enter your GitHub API token.")
+  print("If you don't have one yet, create it at https://github.com/settings/tokens")
+  def get_token():
+    global token
+    data = None
+    while True:
+      if not token:
+        token = getpass.getpass(prompt='Token: ')
+      try:
+        client.inject_token('Bearer ' + token)
+        query = '''
+        query {
+          viewer {
+            login
+          }
+        }
+        '''
+        response = client.execute(query) # This prints the response in case of error, despite any try/except blocks
+        data = json.loads(response)
+        if 'login' in data['data']['viewer']:
+          print(f"Grand, thank you @{data['data']['viewer']['login']}!")
+          print("Do you want learn how to save this token for future use? [y/N]")
+          save = input()
+          if save.lower() == "y":
+            print(textwrap.dedent(f"""
```

Comment:
I'm still not convinced by this huge explanation because it makes it harder to understand the surrounding code (one has to scroll a lot to see what the code does before/after that text block). If you really want to have the explanation, I would move this out of the way here, for example by creating a function that prints the details. This would separate "UI" and "business logic" a bit more.